exe/deprecations

```diff
@@ -85,14 +96,23 @@ end
 option_parser.parse!
 
 options[:mode] = ARGV.last
-path = options[:next] ? "spec/support/deprecation_warning.next.shitlist.json" : "spec/support/deprecation_warning.shitlist.json"
+path = if options[:path]
+  options[:path]
+else
+  options[:next] ? "spec/support/deprecation_warning.next.shitlist.json" : "spec/support/deprecation_warning.shitlist.json"
+end
 
 pattern_string = options.fetch(:pattern, ".+")
 pattern = /#{pattern_string}/
 
-deprecation_warnings = JSON.parse(File.read(path)).each_with_object({}) do |(test_file, messages), hash|
-  filtered_messages = messages.select {|message| message.match(pattern) }
-  hash[test_file] = filtered_messages if !filtered_messages.empty?
+file_paths = Dir.glob(path)
+
+if file_paths.empty?
+  abort "No files found with name #{options[:path]}"
+  exit 2
+else
+  deprecation_warnings = file_paths.map { |path| extract_deprecations(path, pattern) }
+                                   .inject(:merge)
```

Comment:
the `inject(:merge)` logic does not work well with more than 2 files.
I did this:

1. created a folder `dep_test` and added 3 files `dep1.json`, `dep2.json`, and `dep3.json`.

dep1.json

```
{
  "./spec/controllers/contacts_controller_spec.rb": [
    "DEPRECATION WARNING: dep1",
    "DEPRECATION WARNING: dep1"
  ],
  "./spec/controllers/search_controller_spec.rb": [
    "DEPRECATION WARNING: dep2",
    "DEPRECATION WARNING: dep2",
    "DEPRECATION WARNING: dep2"
  ]
}
```

dep2.json
```
{
  "./spec/controllers/search_controller_spec.rb": [ 
    "DEPRECATION WARNING: dep3",
    "DEPRECATION WARNING: dep3"
  ]
}
```

dep3.json
```
{
  "./spec/controllers/contacts_controller_spec.rb": [
    "DEPRECATION WARNING: dep4"
  ],
  "./spec/controllers/search_controller_spec.rb": [
    "DEPRECATION WARNING: dep3",
    "DEPRECATION WARNING: dep2",
    "DEPRECATION WARNING: dep2"
  ]
}
```


Now I run this command
```
exe/deprecations info --path "/home/rishi/projects/next_rails/dep_test/*"
```

I get this output
```
Ten most common deprecation warnings:
Occurrences: 3
DEPRECATION WARNING: dep2
----------
Occurrences: 2
DEPRECATION WARNING: dep1
----------
```

You can see it ignored the deprecation warning `dep4` and `dep3` completely.



I tried to debug this and figured out that the `inject` is not working as expected.
I did something like this (it could be made better for sure):
```
extracted_deprecation_warnings = file_paths.map { |path| extract_deprecations(path, pattern) }

  deprecation_warnings = {}
  extracted_deprecation_warnings.each do |a|
    a.keys.each do |h|
      if deprecation_warnings[h]
        deprecation_warnings[h] << a[h].flatten
        deprecation_warnings[h].flatten!
      else
        deprecation_warnings[h] = a[h].flatten
      end
    end
  end

  deprecation_warnings
```

and this returns
```
rishi@rishi:~/projects/next_rails$ exe/deprecations info --path "/home/rishi/projects/next_rails/dep_test/*"
Ten most common deprecation warnings:
Occurrences: 5
DEPRECATION WARNING: dep2
----------
Occurrences: 3
DEPRECATION WARNING: dep3
----------
Occurrences: 2
DEPRECATION WARNING: dep1
----------
Occurrences: 1
DEPRECATION WARNING: dep4
----------
```

Reply:
I have not tested my suggested code with all possible scenario. I was just trying to suggest an example for you to play with if it seems correct to you.

Reply:
This would be an interesting test case to be added to the test suite.